jsonpb/jsonpb_test.go

```diff
@@ -439,6 +439,18 @@ func TestMarshaling(t *testing.T) {
 	}
 }
 
+func TestMarshalingWithJSONPBMarshaler(t *testing.T) {
+	rawJson := `{ "foo": "bar", "baz": [0, 1, 2, 3] }`
+	msg := notGeneratedMessage{rawJson: rawJson}
+	str, err := new(Marshaler).MarshalToString(&msg)
+	if err != nil {
+		t.Errorf("an unexpected error occurred when marshalling JSONPBMarshaler: %v", err)
+	}
+	if str != rawJson {
+		t.Errorf("Marshalling JSON produced incorrect output\nExpected: %s\nGot: %s", rawJson, str)
```

Comment:
A couple of nits.
(1) Start with lowercase for error message for consistency, i.e. "marshalling ...".  
(2) display got first before want.
"marshalling JSON produced incorrect output: got %s want %s"